src/main/java/org/opensearch/sdk/ExtensionRestRequest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.extensions.rest.RestExecuteOnExtensionRequest;
+import org.opensearch.identity.PrincipalIdentifierToken;
+import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ExtensionRestRequest extends TransportRequest {
+
+    private Method method;
+    private String uri;
+    private PrincipalIdentifierToken principalIdentifierToken;
+
+    public ExtensionRestRequest(Method method, String uri, PrincipalIdentifierToken requesterIdentifier) {
+        this.method = method;
+        this.uri = uri;
+        this.principalIdentifierToken = requesterIdentifier;
+    }
+
+    // TODO Check if this constructor is more useful, If so what happens when request object is null
+    protected ExtensionRestRequest(RestExecuteOnExtensionRequest request) {
+        this.method = request.getMethod();
+        this.uri = request.getUri();
+        this.principalIdentifierToken = request.getRequestIssuerIdentity();
+    }
+
+    public ExtensionRestRequest(StreamInput in) throws IOException {
+        super(in);
+        method = in.readEnum(Method.class);
+        uri = in.readString();
+        principalIdentifierToken = in.readNamedWriteable(PrincipalIdentifierToken.class);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeEnum(method);
+        out.writeString(uri);
+        out.writeNamedWriteable(principalIdentifierToken);
+    }
+
+    public Method method() {
```

Comment:
```suggestion
    public Method getMethod() {
```

Reply:
I specifically requested this be changed to `method()` to align with the same method on `RestRequest`.

Relevant context: [JEP 395](https://openjdk.org/jeps/395).  Since Java 14, "Record" syntax uses the non-Get variant:
> For each component in the header, two members: a public accessor method with the same name and return type as the component, and a private final field with the same type as the component;

The intention is a signal to the coder that the value is "shallowly immutable" (`private final`).   That's the case here.

Reply:
Dan suggested we leave it to `method()`...https://github.com/opensearch-project/opensearch-sdk-java/pull/140#discussion_r975898724